iceoryx_binding_c/include/iceoryx_binding_c/types.h

```diff
@@ -28,27 +28,27 @@ struct iox_ws_storage_t_
 {
     // the value of the array size is the result of the following formula:
     // sizeof(WaitSet) / 8
-    uint64_t do_not_touch_me[133];
+    uint64_t do_not_touch_me[2437];
```

Comment:
what? your new waitset needs 20 times more memory or what does this tell me?

Reply:
The problem originates from the `iox::vector<Trigger>` trigger vector and the condition was previously only 8 bytes in size while the trigger has now a size of 152. Mainly it consists of some uint64_t like triggerId and originHash and pointers to callbacks like remove, hasTriggered, cleanup etc which also contain the this pointer.

In the future I was playing with the thought of having a default typed mempool in the posh runtime where the user can allocate memory on the stack. When we would have this the waitset would shrink again to 133.

Reply:
We discussed once if we should make the waitset a template to have a user defined number of max triggers. I would prefer this for the CPP API. 
THis magic 2437 comes from the waitset and its current MAX_TRIGGERS (or how it is called). When someone changes this max number const in posh_types your C API breaks?

Reply:
@budrus. The C API would not break since I have unit tests in place which are verifying the correct size of those structs. See: https://github.com/ApexAI/iceoryx/blob/iox-%23341-waitset-next-gen/iceoryx_binding_c/test/moduletests/test_types.cpp

Reply:
If we would make the cpp waitset a template with the number of triggers as parameters, could this also be supported by the C waitset?

Reply:
Not directly. We need some kind of hacky crappy workaround to support this.
A clean way would be code generation via cmake maybe but for the moment I would try to avoid this.